### PR TITLE
feature: add pouch kill functionality

### PR DESCRIPTION
**Pull request by xiechengsheng**
Signed-off-by: xiechengsheng <XIE1995@whut.edu.cn>

### Ⅰ. Describe what this PR did
add pouch kill functionality

### Ⅳ. Describe how to verify it
```
$ pouch ps
Name            ID       Status          Created          Image                                            Runtime
foo             c926cf   Up 5 seconds    6 seconds ago    registry.hub.docker.com/library/busybox:latest   runc
$ pouch kill foo
foo
$ pouch ps -a
Name            ID       Status                     Created          Image                                            Runtime
foo             c926cf   Exited (137) 9 seconds     25 seconds ago   registry.hub.docker.com/library/busybox:latest   runc
```

### Ⅴ. Special notes for reviews
NONE

### CODE REVIEW

File: client/container_kill.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	q := url.Values{}
+	q.Set("signal", signal)
+
+	resp, err := client.post(ctx, "/containers/"+name+"/kill", q, nil, nil)
+	if err != nil {
+		return err
+	}
+	ensureCloseReader(resp)
+
+	return err
```
Comment (allencloud) [inline client/container_kill.go]: Here, please explicitly `return nil`.

File: daemon/mgr/container.go [hunk 2]
```diff
@@ -1306,6 +1312,32 @@ func (mgr *ContainerManager) Top(ctx context.Context, name string, psArgs string
...
 
+// Kill sends signal to a container.
+func (mgr *ContainerManager) Kill(ctx context.Context, name string, sig int) error {
+	c, err := mgr.container(name)
+	if err != nil {
+		return err
+	}
+
+	if sig != 0 && !signal.ValidSignalForPlatform(syscall.Signal(sig)) {
+		return fmt.Errorf("signal: %d is not suppoerted", sig)
```
Comment (allencloud) [inline daemon/mgr/container.go]: fmt.Errorf("signal %d is not supported", sig)

File: daemon/mgr/container.go [hunk 3]
```diff
@@ -1306,6 +1312,32 @@ func (mgr *ContainerManager) Top(ctx context.Context, name string, psArgs string
...
+	}
+
+	if !c.IsRunning() {
+		return fmt.Errorf("container: %s is not running, can not execute kill command", c.ID)
```
Comment (allencloud) [inline daemon/mgr/container.go]: please remove `:` in the error message.

File: apis/server/container_bridge.go [hunk 4]
```diff
@@ -397,3 +400,24 @@ func (s *Server) waitContainer(ctx context.Context, rw http.ResponseWriter, req
...
+func (s *Server) killContainer(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	name := mux.Vars(req)["name"]
+
+	var sig syscall.Signal
+
+	// parse client signal
+	if sigStr := req.FormValue("signal"); sigStr != "" {
+		var err error
+		if sig, err = signal.ParseSignal(sigStr); err != nil {
+			return err
```
[the remark this answers is not in the document]
Comment (xiechengsheng) [reply apis/server/container_bridge.go]: ok, will add http 400 err here.

**Comment by xiechengsheng** [comment]
@allencloud I have updated codes according to your comments, PTAL, thanks. 😄

**Comment by allencloud** [comment]
@xiechengsheng 
CI fails with:
```
----------------------------------------------------------------------
FAIL: /go/src/github.com/alibaba/pouch/test/api_container_kill_test.go:92: APIContainerKillSuite.TestKillContainerWithInvalidSignal
/go/src/github.com/alibaba/pouch/test/api_container_kill_test.go:112:
    CheckRespStatus(c, resp, 204)
/go/src/github.com/alibaba/pouch/test/util_api.go:25:
    c.Assert(resp.StatusCode, check.Equals, status, check.Commentf("Error:%s", got.Message))
... obtained int = 500
... expected int = 204
... Error:process id fa1602b47d813085d88165f71bcf854abda35163c2d450000ac1156348ab4d6b: not found
```

File: apis/swagger.yml [hunk 5]
```diff
@@ -551,6 +551,26 @@ paths:
...
+      parameters:
+        - $ref: "#/parameters/id"
+        - name: "signal"
+          in: "query"
+          description: "signal to send to the container"
+          type: "string"
+          default: "SIGKILL"
+      responses:
+        204:
+          description: "no error"
```
Comment (allencloud) [inline apis/swagger.yml]: You add status code 400 in code, then here we need to add 400 as well in swagger. @xiechengsheng

**Comment by xiechengsheng** [comment]
OK, I will fix this ASAP.

**Comment by xiechengsheng** [comment]
Wow, merged so quickly. 😄

**Comment by wangforthinker** [comment]
Could we continue to push this feature? @fuweid @xiechengsheng

**Pull request merged**